src/query/sql/src/planner/binder/table.rs

```diff
@@ -565,9 +573,191 @@ impl Binder {
                     .await
             }
             TableReference::Join { .. } => unreachable!(),
+            TableReference::Values {
+                span,
+                values,
+                alias,
+            } => self.bind_values(bind_context, *span, values, alias).await,
         }
     }
 
+    #[async_backtrace::framed]
+    pub(crate) async fn bind_values(
+        &mut self,
+        bind_context: &BindContext,
+        span: Span,
+        values: &Vec<Vec<AExpr>>,
+        alias: &Option<TableAlias>,
+    ) -> Result<(SExpr, BindContext)> {
+        if values.is_empty() {
+            return Err(ErrorCode::SemanticError(
+                "Values lists must have at least one row".to_string(),
+            )
+            .set_span(span));
+        }
+        let same_length = values.windows(2).all(|v| v[0].len() == v[1].len());
+        if !same_length {
+            return Err(ErrorCode::SemanticError(
+                "Values lists must all be the same length".to_string(),
+            )
+            .set_span(span));
+        }
+
+        let num_rows = values.len();
+        let num_cols = values[0].len();
+
+        let mut names = match alias {
+            Some(alias) => {
+                if alias.columns.len() > num_cols {
+                    return Err(ErrorCode::SemanticError(format!(
+                        "table {} has {} columns available but {} columns specified",
+                        alias.name,
+                        num_cols,
+                        alias.columns.len()
+                    ))
+                    .set_span(span));
+                }
+                alias
+                    .columns
+                    .iter()
+                    .map(|c| c.to_string())
+                    .collect::<Vec<_>>()
+            }
+            None => vec![],
+        };
+        // For columns whose names are not specified, assigns default column names col0, col2, etc.
+        for i in names.len()..num_cols {
+            names.push(format!("col{}", i));
+        }
+
+        let mut bind_context = BindContext::with_parent(Box::new(bind_context.clone()));
```

Comment:
Seems there is no need to create a new `bind_context`.

Reply:
Can we use `let mut bind_context = bind_context.clone();`? Because we need to return `BindContext`, but the original `bind_context` is `&BindContext`

Reply:
https://github.com/datafuselabs/databend/pull/12380#discussion_r1288196126